core/src/main/java/org/neo4j/graphalgo/core/utils/RelationshipContainer.java

```diff
@@ -0,0 +1,215 @@
+            return null;
+        }
+    };
+
+    private final int[][] data;
+
+    private RelationshipContainer(int nodeCount) {
+        data = new int[nodeCount][];
+    }
+
+    /**
+     * called for each relationship for nodeId
+     * @param nodeId the source nodeid
+     * @param consumer the relation consumer
+     */
+    public void forEach(int nodeId, RelationshipConsumer consumer) {
+        if (nodeId >= data.length) {
+            return;
+        }
+        final int[] relationships = data[nodeId];
+        for (int relationship : relationships) {
+            consumer.accept(nodeId, relationship, -1L);
+        }
+    }
+
+    /**
+     * get an iterator for each relation
+     */
+    public Iterator<RelationshipCursor> iterator(int nodeId) {
+        if (nodeId >= data.length) {
+            return EMPTY_IT;
+        }
+        return new RelationIterator(data, nodeId);
+    }
+
+    /**
+     * get RelationContainer.Builder to ease construction of the container
+     * @param nodeCount the nodeCount
+     * @return a Builder for RelationContainers
+     */
+    public static Builder builder(int nodeCount) {
+        return new Builder(nodeCount);
+    }
+
+    /**
+     * get an Importer object to ease the import from neo4j
+     * @param api the core api
+     * @return an Importer for RelationContainer
+     */
+    public static RCImporter importer(GraphDatabaseAPI api) {
+        return new RCImporter(api);
+    }
+
+    private static class RelationIterator implements Iterator<RelationshipCursor> {
+
+        private final RelationshipCursor cursor;
+        private final int[] relationships;
+        private final int size;
+        private int offset = 0;
+
+        private RelationIterator(int[][] data, int sourceNodeId) {
+            this.relationships = data[sourceNodeId];
+            size = relationships.length;
+            cursor = new RelationshipCursor();
+            cursor.sourceNodeId = sourceNodeId;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return offset < size;
+        }
+
+        @Override
+        public RelationshipCursor next() {
+            cursor.targetNodeId = relationships[offset++];
+            return cursor;
+        }
+    }
+
+    public static class Builder {
+
+        private final RelationshipContainer container;
+
+        private int nodeId;
+        private int offset;
+
+        private Builder(int nodeCount) {
+            container = new RelationshipContainer(nodeCount);
+        }
+
+        /**
+         * set the builder to aim for nodeId and initialize its underlying connection array
+         *
+         * @param nodeId the current (mapped) nodeId
+         * @param degree the degree for the node
+         * @return this instance for method chaining
+         */
+        public Builder aim(int nodeId, int degree) {
+            this.nodeId = nodeId;
+            offset = 0;
+            container.data[nodeId] = new int[degree];
```

Comment:
Or perhaps we find an even better struct for this case. I made issue #69 out of it but im going to delay it until we need it.